src/components/Card.js

```diff
@@ -1,30 +1,43 @@
-export class Card {
-  constructor({ name, link }, cardSelector, handleImageClick) {
+import confirmPopup from "../pages/index.js";
+
+class Card {
+  constructor(
+    { name, link, _id, isLiked },
+    cardSelector,
+    handleImageClick,
+    handleLikeToggle,
+    handleCardDelete
+  ) {
     this._name = name;
     this._link = link;
+    this._isLiked = isLiked;
+    this._id = _id;
     this._cardSelector = cardSelector;
     this._handleImageClick = handleImageClick;
+    this._handleLikeToggle = handleLikeToggle;
+    this._handleCardDelete = handleCardDelete;
   }
 
+  // Card template creation and copying
   _getCardTemplate() {
-    // Here goes the method for card template creation and copying
     const cardTemplate = document
       .querySelector(this._cardSelector)
       .content.firstElementChild.cloneNode(true);
     return cardTemplate;
   }
 
+  // Listeners creation function
   _setEventListeners() {
-    //Here goes the part of code, responsible for the event listeners
-
-    //like button
     this._likeButton.addEventListener("click", () => {
-      this._handleLikeButton();
+      this._handleLikeToggle(this._isLiked); // должен передавать в index.js статус isLiked и id,
     });
 
-    //delete button
+    //Delete button
     this._trashButton.addEventListener("click", () => {
-      this._deleteButtonHandler();
+      confirmPopup.open();
+      confirmPopup._handleFormSubmit = () => {
```

Comment:
`_handleFormSubmit`

Please, delete the **low dash** (`underscore`) from methods/fields if you want to use them **outside** a class because the `low dash` shows a **private** field which is used only inside the class.